File: leetcode/283.移动零/283.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slow = 0
        for fast in range(0,len(nums)):
            if nums[fast] != 0:
                nums[slow] = nums[fast]
                slow = slow + 1
        for i in range(slow, len(nums)):
            nums[i] = 0
        

File: leetcode/283.移动零/283.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int slow = 0;
        for (int fast = 0; fast < nums.size(); fast++) {
            if (nums[fast]) {
                nums[slow++] = nums[fast];
            }
        }
        for (int i = slow; i < nums.size(); i++) {
            nums[i] = 0;
        }
    }
};